platformio.ini

```diff
@@ -335,6 +335,31 @@ lib_ignore    = Adafruit NeoPixel, TMCStepper
 src_filter    = ${common.default_src_filter} +<src/HAL/HAL_STM32_F4_F7> -<src/HAL/HAL_STM32_F4_F7/STM32F7>
 monitor_speed = 250000
 
+#
+# FYSETC S6 (STM32F446VET6 ARM Cortex-M4)
+#
+[env:FYSETC_S6]
+platform      = ststm32@5.6.0
+framework     = arduino
+board         = fysetc_s6
+extra_scripts = buildroot/share/PlatformIO/scripts/fysetc_STM32S6.py
+build_flags   = ${common.build_flags}
+  -DTARGET_STM32F4 -std=gnu++14
+  -DVECT_TAB_OFFSET=0x10000
+  -DUSBCON -DUSBD_USE_CDC -DHAL_PCD_MODULE_ENABLED -DUSBD_VID=0x0483 '-DUSB_PRODUCT="FYSETC_S6"'
+build_unflags = -std=gnu++11
+lib_deps      = ${common.lib_deps}
+  SoftwareSerialM=https://github.com/FYSETC/SoftwareSerialM/archive/master.zip
+lib_ignore    = c1921b4
+  TMC26XStepper
```

Comment:
SoftwareSerialM is incompatible with #15655 

If this change is merged after #15655, `SoftwareSerialM` needs to be removed, and `SoftwareSerial` (From Arduino_Core_STM32) needs to be ignored.

Reply:
If 15655 is merged before this PR, we can add more commits to this PR's branch as needed to get conflicts resolved and resolve these issues.

Reply:
i use STM 5.6.0 for the FYSETC S6 , so it won't conflict with the SoftwareSerial in STM 5.7.0. But we can wait for the STM 5.7.0 to be merged and test then resolve conflicts as thinkyhead said .I will keep these PR in my own branch for FYSETC customers.

Reply:
5.7.0 is now active, so if you see no build conflicts then there may be nothing to do.